### ISSUE TITLE
Dynamic model widget shows unfiltered value

### ISSUE THREAD

**Issue by apishark** [netbox-community/netbox#12189]
### NetBox version

v3.4.4

### Python version

3.10

### Steps to Reproduce
Browse to http://localhost:8001/dcim/devices/add/
Select a Manufacturer
Select a Device type
Select another Manufacturer

### Expected Behavior
Device type shows filtered value

### Observed Behavior
Device type shows unfiltered value

![image](https://user-images.githubusercontent.com/98494314/230364931-9e8be3a1-e427-4996-9145-563fbdeb683c.png)
![image](https://user-images.githubusercontent.com/98494314/230364986-e8c0d246-61c7-4b17-8314-6850a080edb8.png)

**Comment by jeremystretch** [comment]
Could you please clean up your description above? It appears you've pasted the same content twice under "steps to reproduce" and "expected behavior."

**Comment by apishark** [comment]
One is UNfiltered and the other is filtered. 
In plain English, an unfiltered value is shown in the Device Type, which is not the expected behavior.

**Comment by jeremystretch** [comment]
Ok, please update your description above to reflect that, rather the duplicating the reproduction steps in their entirety.

**Comment by apishark** [comment]
NetBox version
v3.4.4

Python version
3.10

Steps to Reproduce
Browse to http://localhost:8001/dcim/devices/add/
Select a Manufacturer
Select a Device type
Select another Manufacturer
Device type shows unfiltered value

Expected Behavior
Device type shows filtered value

Observed Behavior
![image](https://user-images.githubusercontent.com/98494314/230630840-a9626cd4-56f0-4b72-a5b1-46845517ad4d.png)
![image](https://user-images.githubusercontent.com/98494314/230630858-2956ba45-037d-494d-b9a9-d132841f0554.png)

**Comment by kkthxbye-code** [comment]
The manufacturer field does not exist in the device creation for in 3.5. The devicetype now has the advanced object selection field. As such there is nothing to fix here.

Also I'm not sure what the initial issue was. When we filter a DynamicModelChoiceField based on another DynamicModelChoice field we do not clear the currently selected value, but the filter is still applied to other non-selected values. I'm not sure if we wanted to change that, as it's pretty consistent throughout the UI.
